hangman.py

```diff
@@ -0,0 +1,300 @@
+        tries = vowel_check(letter, tries)
+        print("Oops! That letter is not in my word:", word)
+    letters_guessed.add(letter)
+    return tries, warn, letters_guessed, word
+
+
+def start_message(secret_word, hints):
+    """
+        secret_word: string, the word the user is guessing
+        hints: boolean, hints are turned on or off
+        This function shows a starting message.
+    """
+    print("Welcome to the game Hangman!")
+    print("I am thinking of a word that is", len(secret_word), "letters long.")
+    print("You have 6 warnings left.")
+    if hints:
+        print('You have also 1 hint. Enter "*" to use it.')
+    return
+
+
+def hangman(secret_word, hints=False):
+    """
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    """
+    tries = INITIAL_TRIES
+    warn = INITIAL_WARN
+    # List "letters_guessed" is used as a container for letters which have already been guessed
+    letters_guessed = set()
+    # Variable "word" shows the length of the secret word and which letters have already been guessed
+    word = get_guessed_word(secret_word, letters_guessed)
+    start_message(secret_word, hints)
+    while tries != 0 and is_word_guessed(secret_word, word):
+        print("-" * 13)
+        print("You have", tries, "guesses left.")
+        print("Available letters:", get_available_letters(letters_guessed))
+        letter = str(input("Please guess a letter: ")).lower()
+        if len(letter) == 1:
+            tries, warn, letters_guessed, word = check_element(tries, warn, letters_guessed, word, letter, hints)
+
+        # If user enters more or less than one symbol, the program issues a warning
+        else:
+            warn, tries = count(warn, tries)
+            print("Oops! Don't enter more or less than one symbol. You have", *[warn if warn >= 0 else "no"],
+                  "warnings left" + str(*[" so you lose one guess" if warn < 0 else ""]) + ":", word)
+
+    print("-" * 13)
+    if tries == 0:
+        print('Sorry, you ran out of guesses. The word was "' + secret_word + '".')
+    else:
+        print("Congratulations, you won! Your total score for this game is:", tries * len(set(secret_word)))
```

Comment:
I see you created a separate function for `start_message`, so I believe it makes sense to do the same with these `end_message` as well